ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/BigDiffy.scala

```diff
@@ -541,6 +541,16 @@ object BigDiffy extends Command with Serializable {
     }
   }
 
+  // a null TableFieldSchema mode can be treated as "NULLABLE", which is the
+  // default value according to the docs
+  private def getNormalizedFieldMode(f: TableFieldSchema): String = {
+    if (f.getMode == null) {
```

Comment:
it's unclear to me what scenarios the BigQuery API returns `null` vs `NULLABLE`, but different implementations treat this differently. For example the GCP console shows "NULLABLE" for both cases 
![image](https://github.com/spotify/ratatool/assets/18561476/b9795680-09e8-4596-a98e-af877bda29e1)

however the `bq` CLI doesn't show any mode for nullable fields

```
<~>-2-> bq show telemetry-playground:staging.federicac_page_aggregate_daily_20240101
Table telemetry-playground:staging.federicac_page_aggregate_daily_20240101

   Last modified                Schema                Total Rows   Total Bytes     Expiration      Time Partitioning   Clustered Fields   Total Logical Bytes   Total Physical Bytes   Labels
 ----------------- --------------------------------- ------------ ------------- ----------------- ------------------- ------------------ --------------------- ---------------------- --------
  05 Feb 07:12:41   |- partition_timestamp: integer   15518727     3298969346    06 Mar 07:12:41                                          3298969346            3815264611
                    |- client_id: string
                    |- os_name: string
                    |- correlation_id: string
                    |- user_id: string (required)
                    |- device_type: string
                    |- page_id: string
                    |- navigational_root: string
                    |- app_version: string
                    |- foreground_time: integer
                    |- page_id_count: integer

<~>-> bq show foreground-aggregates:page_interval_daily_aggregate_sample.page_interval_daily_aggregate_sample_v1_20240101
Table foreground-aggregates:page_interval_daily_aggregate_sample.page_interval_daily_aggregate_sample_v1_20240101

   Last modified                Schema                Total Rows   Total Bytes     Expiration      Time Partitioning   Clustered Fields   Total Logical Bytes   Total Physical Bytes   Labels
 ----------------- --------------------------------- ------------ ------------- ----------------- ------------------- ------------------ --------------------- ---------------------- --------
  05 Feb 09:57:02   |- partition_timestamp: integer   15518727     3298969346    02 Jan 22:29:20                                          3298969346            2290031265
                    |- client_id: string
                    |- os_name: string
                    |- correlation_id: string
                    |- user_id: string (required)
                    |- device_type: string
                    |- page_id: string
                    |- navigational_root: string
                    |- app_version: string
                    |- foreground_time: integer
                    |- page_id_count: integer
```

![image](https://github.com/spotify/ratatool/assets/18561476/a34db741-9b58-4b02-b1cf-83909b1120a6)

Reply:
although I'm surprised this is the first time something like this has come up, nullable fields are very common in BQ, so I would expect this to have come up earlier

Reply:
If we need to merge this to unblock, can we at least open a ticket with google about this. I don't believe we should litter the codebase with google api semantics

Reply:
https://partnerissuetracker.corp.google.com/issues/324053124

Reply:
@idreeskhan I was able to replicate this by creating a BQ table via a BQ SQL CREATE query, fields marked as nullable have a null mode in the BQ Java API rather than the correct value of NULLABLE